trackintel/preprocessing/positionfixes.py

```diff
@@ -1,512 +1,572 @@
+            started_at=("tracked_at", "min"),
+            finished_at=("tracked_at", "max"),
+            geom=(pfs.geometry.name, lambda x: LineString(x.tolist())),
+        )
+        tpls = tpls.set_geometry("geom")
+        tpls.crs = pfs.crs
+    elif method == "overlap_staypoints":
+        tpls = _generate_triplegs_overlap_staypoints(cond_gap, pfs, staypoints)
+    else:
+        raise ValueError(
+            f"Method unknown. We only support 'between_staypoints' and 'overlap_staypoints'. You passed {method}"
+        )
+
+    # assert validity of triplegs
+    tpls, pfs = _drop_invalid_triplegs(tpls, pfs)
+
+    if not staypoints_exist:
+        pfs.drop(columns="staypoint_id", inplace=True)
+
+    # dtype consistency
+    pfs["tripleg_id"] = pfs["tripleg_id"].astype("Int64")
+    tpls.index = tpls.index.astype("int64")
+    tpls.index.name = "id"
+
+    # user_id of tpls should be the same as pfs
+    tpls["user_id"] = tpls["user_id"].astype(pfs["user_id"].dtype)
+    if len(tpls) == 0:
+        warnings.warn("No triplegs can be generated, returning empty tpls.")
+        return pfs, tpls
+
+    return pfs, Triplegs(tpls)
+
+
+def _generate_triplegs_overlap_staypoints(cond_gap, pfs, staypoints):
+    """Connect staypoints with overlapping triplegs
+
+    Parameters
+    ----------
+    cond_gap : A boolean mask indicating gaps in the pfs data frame.
+    pfs : Positionfixes
+    staypoints : Staypoints
+
+    Returns
+    -------
+    tpls: Triplegs
+        tpls with geometries.
+
+    Notes
+    -----
+    In case of a staypoint with only one positionfix the previous tripleg will have a spatial overlap with the
+    staypoint, while the following tripleg will not overlap with the staypoint. Otherwise, the positionfix of the
+    staypoints would need a duplication.
+    """
+    # initialize: keep original ids
+    tpls_ids = pfs["tripleg_id"].copy()
+
+    # conditions to identify overlap positions in the positionfixes:
+    # not a new user & not a tripleg & not a gap at staypoint start & not a gap at staypoint end
+    cond_not_tpl = ~pd.isna(pfs["staypoint_id"])
+    cond_overlap = ~(pfs["user_id"] != pfs["user_id"].shift(1)) & cond_not_tpl
+
+    # temporal overlap: overlap tripleg end with start of next staypoint
+    cond_overlap_start = cond_overlap & ~cond_gap & pd.isna(pfs["tripleg_id"])
+    pfs.loc[cond_overlap_start, "tripleg_id"] = tpls_ids.shift(1)[cond_overlap_start]
+    tpls = pfs.groupby("tripleg_id").agg(
+        user_id=("user_id", "first"), started_at=("tracked_at", "min"), finished_at=("tracked_at", "max")
+    )
+
+    # spatial overlap: overlap tripleg with the location of previous and next staypoint
+    cond_overlap_end = cond_overlap & ~cond_gap.shift(-1).fillna(False) & pd.isna(pfs["tripleg_id"])
+    pfs.loc[cond_overlap_end, "tripleg_id"] = tpls_ids.shift(-1)[cond_overlap_end]
+    cond_empty = pd.isna(pfs["tripleg_id"])
+    pfs.loc[cond_empty, "tripleg_id"] = tpls_ids[cond_empty]

```

Comment:
Minor thing: can we maybe put a comment here to explain the process a bit better? It took me quite a while to understand that the `.shift(1)` and `.shift(-1)` introduce NaNs in places where we had a tripleg ID before. Maybe we can rename `tpls_ids` to `between_tpls_ids` or `initial_tpls_ids` to make clear that these are the tripleg IDs that are generated with the between-method.

Reply:
Good point, the variable is renamed to `between_tpls_ids` to clarify its origin and the comment is adjusted.